src/assertions.ts

```diff
@@ -1296,6 +1305,86 @@ function containsJSON(str: string): any {
   return jsonObjects;
 }
 
+export async function isSql(
+  outputString: string,
+  renderedValue: AssertionValue | undefined,
+  inverse: boolean,
+  assertion: Assertion,
+): Promise<GradingResult> {
+  let pass = false;
+  let parsedSql;
+  let databaseType: string = 'MySQL';
+  let whiteTableList: string[] | undefined;
+  let whiteColumnList: string[] | undefined;
+
+  if (renderedValue && typeof renderedValue === 'object') {
+    const value = renderedValue as {
+      database?: string;
+      whiteTableList?: string[];
+      whiteColumnList?: string[];
+    };
+
+    databaseType = value.database || 'MySQL';
+    whiteTableList = value.whiteTableList;
+    whiteColumnList = value.whiteColumnList;
+  }
+
+  if (renderedValue && typeof renderedValue !== 'object') {
+    throw new Error('is-sql assertion must have a object value.');
+  }
+
+  const { Parser: SqlParser } = await import('node-sql-parser').catch(() => {
+    throw new Error('node-sql-parser is not installed. Please install it first');
+  });
+
+  const sqlParser = new SqlParser();
+
+  const opt: sqlParserOption = { database: databaseType };
+
+  const failureReasons: string[] = [];
+
+  try {
+    parsedSql = sqlParser.astify(outputString, opt);
+    pass = !inverse;
+  } catch (err) {
+    pass = inverse;
+    failureReasons.push(
+      `SQL statement does not conform to the provided ${databaseType} database syntax.`,
+    );
+  }
+
+  if (whiteTableList) {
+    opt.type = 'table';
+    try {
+      sqlParser.whiteListCheck(outputString, whiteTableList, opt);
+    } catch (err) {
+      pass = inverse;
+      failureReasons.push('It failed the provided authority table list check.');
```

Comment:
Is there useful information in the error that we can pass along? 
 Ideally something like "SQL output references disallowed table X"

Reply:
When the LLM output string is `UPDATE employees SET department_id = 2 WHERE employee_id = 1`, and the allowedTables is `['(select|update|insert|delete)::null::departments']`

The output error is something like this: 
 Error: authority = 'update::null::employees' is required in table whiteList to execute SQL = 'UPDATE employees SET department_id = 2 WHERE employee_id = 1'
        at r.whiteListCheck (\Users\taozhi\Workspaces\OpenSources\node-sql-parser\src\parser.js:57:26)
        at isSql (D:\promptfoo place\promptfoo\src\assertions.ts:1359:17)
        ....
        
        So basically if the output string tries to do an operation that is not specified(allowed) in the list, the error will report what is required in the table whiteList to execute it. Something like this: `'delete::null::employees' is required in table whiteList to execute SQL = 'DELETE FROM employees;'`

Reply:
I think that is ok.  I would make the full reason something like: `SQL validation failed: <error>`

Reply:
updated!